src/Behat/Behat/Snippet/Generator/SnippetGenerator.php

```diff
@@ -26,21 +26,11 @@ interface SnippetGenerator
 {
     /**
      * Checks if generator supports search query.
-     *
-     * @param Environment $environment
-     * @param StepNode    $step
-     *
-     * @return bool
      */
-    public function supportsEnvironmentAndStep(Environment $environment, StepNode $step);
+    public function supportsEnvironmentAndStep(Environment $environment, StepNode $step): bool;
```

Comment:
Adding these return types in the interface is a BC break. If an external extension implements this interface and they have not added these return types, php will throw an error because the declaration of these functions in their classes does not match the declaration in the interface, so please remove them
It is ok to leave the return types in the `ContextSnippetGenerator` class because adding more types in a derived or implementing class is OK and the class is final, so there can't be any other class out there extending it